Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,578 @@
+
+You can also specify a **route prefix** when registering route components. This is a string that will prefix all the paths on this API. There's an `AddRouteComponents` overload that accepts a route prefix as the first argument:
+
+```c#
+services.AddControllers().AddOdata(
+    options => options.AddRouteComponents("odata", edmModel));
+```
+In this case, paths will have the `odata` prefix, e.g. `/odata/Customers`, `/odata/Products`, `/odata/$metadata`.
+
+The application's base URL together with the route prefix form the **service URL** of the OData service. For example, if the application's base URL is `https://service.example.org` and the route prefix is `odata`, then the service URL for this OData service will be `https://service.example.org/odata`. If the route prefix is empty (i.e. default route), then the service URL will be `https://service.example.org`.
+
+You can call the `AddRouteComponents()` method multiple times with different prefixes (and possibly different `IEdmModel` instances) to serve multiple OData APIs in the same application:
+
+```c#
+services.AddControllers().AddOData(
+    options => options.AddRouteComponents("api1", edmModel1).AddRouteComponents("api2", edmModel2));
+```
+
+This will create OData services with the following url:
+- `<baseUrl>/api1` using the schema defined by `edmModel1`.
+- `<baseUrl>/api2` using the schema defined by `edmModel2`.
+
+## Routing
+
+> [!IMPORTANT]
+> ASP.NET Core OData routing is based on controllers and therefore does not support [minimal APIs](/aspnet/core/fundamentals/minimal-apis/overview).
+
+The OData specification defines a number of conventions for routing and endpoints for accessing various resources based on REST best-practices. For example, if you have an entity set called `Customers` based on the `Customer` entity type, the service will be expected to expose a `GET /Customers` endpoint that returns a collection of `Customer` entities, it will also be expected to expose a `GET /Customers({id})` endpoint that returns a customer based on ID.
+
+The ASP.NET Core OData library makes it easy to create endpoints that match OData's routing conventions. Routing in ASP.NET Core OData is based on controllers. For example, here's what a controller for the `Customers` endpoint could look like:
+
+```c#
+public class CustomersController : ODataController
+{
+    DbContext _db;
+
+    public CustomersController(DbContext db)
+    {
+        _db = db;
+    }
+
+    public ActionResult<IQueryable<Customer>> Get()
+    {
+        return Ok(_db.Customers);
+    }
+
+    public ActionResult<Customer> Get(int id)
+    {
+        return _db.Customers.Find(id);
+    }
+
+    public ActionResult<Customer> Post(Customer customer)
+    {
+        _db.Customers.Add(customer);
+        _db.SaveChanges();
+        return Created(customer);
+    }
+}
+```
+
+- The `Get()` controller action will be mapped to the `GET /Customers` endpoint.
```

Comment:
I guess in general API jargon, endpoints refer to the URLs/paths that clients can hit to request a resource (e.g. [see this discussion](https://stackoverflow.com/questions/18929325/what-is-meant-by-an-api-endpoint)). I think the term is overloaded, and ASP.NET Core may have other uses of the term. Since I have not used the term `route template` before in this article, maybe I can use `route` instead if `endpoint` is ambiguous.

Reply:
The endpoint mapping table doesn't specify what is an endpoint (I mean the term "endpoint" doesn't appear in any of the table headings).